tools/sklearn/dict_h5_pickler.py

```diff
@@ -0,0 +1,510 @@
+
+        # nested method for recursion
+        def recursive_save_model(h5file_obj, dictionary):
+            for model_key, model_value in dictionary.items():
+                type_name = type(model_value).__name__
+                try:
+                    if type_name in ['list']:
+                        if len(model_value) > 0:
+                            list_obj = all(isinstance(x, dict) for x in model_value)
+                            if list_obj is False:
+                                self._create_dataset(h5file_obj, model_key, model_value)
+                            else:
+                                for index, model_item in enumerate(model_value):
+                                    model_key_item = model_key + "/" + str(index)
+                                    if model_item is not None:
+                                        if model_key_item in h5file_obj:
+                                            recursive_save_model(model_key_item, model_item)
+                                        else:
+                                            group = h5file_obj.create_group(model_key_item)
+                                            recursive_save_model(group, model_item)
+                                    else:
+                                        self._create_dataset(h5file_obj, model_key_item, model_item)
+                        else:
+                            self._create_dataset(h5file_obj, model_key, model_value)
+                    elif type_name in ['dict']:
+                        if isinstance(model_key, (int, numpy.int16, numpy.int32, numpy.int64)) is True:
+                            model_key = str(model_key)
+                        if model_key in h5file_obj:
+                            recursive_save_model(h5file_obj[model_key], model_value)
+                        else:
+                            group = h5file_obj.create_group(model_key)
+                            recursive_save_model(group, model_value)
+                    else:
+                        self._create_dataset(h5file_obj, model_key, model_value)
+                except Exception:
+                    continue
+        recursive_save_model(h5file, serialised_model)
+
+    def _restore_list_in_model(self, model_object):
+        """
+        Convert the model dictionary to list if the keys are represented as numbers
+        """
+        if isinstance(model_object, dict) is True:
+            for key, value in model_object.items():
+                if type(value).__name__ in ['dict']:
+                    keys = value.keys()
+                    all_keys_number = all(str.isnumeric(x) for x in keys)
+                    if all_keys_number is True:
+                        keys = [int(ky) for ky in keys]
+                        model_object[key] = list()
+                        for idx in range(len(keys)):
+                            model_object[key].append(value[str(idx)])
```

Comment:
How about some keys are numeric, some are not?

Reply:
Lists containing dicts with the same set of keys. All these dicts are stored a big dict with their indices as their keys.